modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -2525,22 +2525,31 @@ private GroupByStep _getGroupByStep(
 			CommerceOrderTable.INSTANCE.groupId.eq(groupId)
 		);
 
-		if (ArrayUtil.isNotEmpty(commerceAccountIds)) {
-			predicate = predicate.and(
-				CommerceOrderTable.INSTANCE.commerceAccountId.in(
-					ArrayUtil.toArray(commerceAccountIds)));
+		if (commerceAccountIds != null) {
+			if (commerceAccountIds.length == 0) {
+				predicate = predicate.and(
+					CommerceOrderTable.INSTANCE.commerceAccountId.isNull());
+			}
+			else {
+				predicate = predicate.and(
+					CommerceOrderTable.INSTANCE.commerceAccountId.in(
+						ArrayUtil.toArray(commerceAccountIds)));
+			}
```

Comment:
The logic check is required to mimic how we return results in our [CommerceOrderPreFilterContributor](https://github.com/liferay/liferay-portal/blob/c9150aed3be5328d444c4f9c5b4f6a60a905f32f/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/search/spi/model/query/contributor/CommerceOrderModelPreFilterContributor.java#L73). If `commerceAccountIds` is null, we return all results without filtering. If `commerceAccountIds` is of size 0, we filter by commerceAccountId == null, effectively returning no results.

This same logic pattern applies to orderStatuses as well.